pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -113,6 +113,9 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
 		defaultConfig, err := DefaultClientConfig.ClientConfig()
+		if IsConfigurationInvalid(err) {
```

Comment:
This is correct - the original Kube code here never had a non-default case, so we simply overlooked it.

Please add a test to merged_client_builder_test.go that expresses this code path.